lib/dep_graph.js

```diff
@@ -74,7 +79,25 @@ DepGraph.prototype = {
    * Check if a node exists in the graph
    */
   hasNode:function (node) {
-    return !!this.nodes[node];
+    return this.nodes.hasOwnProperty(node);
+  },
+  /**
+   * Get the data associated with a node name
+   */
+  getNodeData:function (node) {
+    if (this.hasNode(node)) {
+      return this.nodes[node];
+    } else {
+      throw new Error('Node does not exist: ' + node);
+    }
+  },
+  /**
+   * Set the associated data for a given node name. If the node does not exist, this method will do nothing
+   */
+  setNodeData:function (node, data) {
```

Comment:
I feel like this should also throw an error if the node doesn't exist. `setNodeData` shouldn't be called on non-existent nodes.